src/Linters/LintRule.hack

```diff
@@ -0,0 +1,22 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST;
+
+/**
+ * An abstract lint rule that provides a single error reason, which could be
+ * either a HHAST linter or a lint rule written in OCaml.
+ */
+<<__Sealed(
+  SingleRuleLinter::class,
+  // HHClientLintRule::class,
+)>>
+interface LintRule {
+  public function getName(): string;
```

Comment:
- add a doc comment here explaining that it's used for suppression codes and showing for users
- perhaps add a sepaate 'getSuppressionCode', with default implementations in the base classes that return getName()

Reply:
Since the suppression logic is in the linter, we don't have to create `getSuppressionCode` here. Instead we can provide a trait for implementing Linters filtering out lint errors.

Anyway, we have chance to add more methods into `LintRule`. This PR aims minimize the API changes, especially when `LintRule` is sealed, allowing adding methods into it while keeping backward compatibility.

Reply:
> Since the suppression logic is in the linter

I think this makes sense for SingleRuleLinters, but as we want separate suppression codes per lint rule, they should probably be here, or separately in the lint error. The suppressio ncode for hhclientlinter will vary by concrete rule/error.

Reply:
`getErrorCode` added